twisted/twisted
# 11564 Update the development documentation to cover the latest PR process.

## Scope and purpose

Fixes #11564

This is a big PR as our current development documentation is a bit neglected... I would say it's chaos

I have removed this page https://docs.twisted.org/en/latest/core/development/policy/code-dev.html as the title is "Working from Twisted’s code repository" but then talks about emacs and debian pages.

The plan is to move all the pages from Policy (https://docs.twisted.org/en/latest/core/development/policy/index.html) into the root of "Development of Twisted" (https://docs.twisted.org/en/latest/core/development/index.html)
I think that all this extra hierarchy is not needed.

This is why the new pages were created directly into `docs/core/development/`.

We might even want to have them one level up as `docs/development/`.

The base `CONTRIBUTING.md` was updated to link to the new dev-process page and not mention Trac. It should be a very brief introduction into the dev process.

The content from `code-dev.rst` was moved to serve as the base for 'docs/core/development/dev-process`
This page should contain more info about the dev process,
But might contain links to separate pages dedicated to testing or the review process.


The full content of the wiki page for review process (https://github.com/twisted/trac-wiki-archive/blob/trunk/ReviewProcess.mediawiki) was move into a separate file.
This can contain all the details of the review process.

## Todo in separate PRs

* [x] #11567 See how to handle the release regression queue/list/report
* [x] #11567 Look at writing code about filling issues. Maybe this will solve the above. 
* [x] #11567 Look at creating issue templates. Might help with the above :)
* [x] #11574 See how to handle merge reverts in trunk 


## How to test

* Check redirected page https://twisted--11566.org.readthedocs.build/en/11566/core/development/policy/code-dev.html
* Check new dev process page https://twisted--11566.org.readthedocs.build/en/11566/core/development/dev-process.html
* Check  the review process page https://twisted--11566.org.readthedocs.build/en/11566/core/development/review-process.html


## Contributor Checklist:

* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://docs.twistedmatrix.com/en/stable/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://github.com/twisted/trac-wiki-archive/blob/trunk//ReviewProcess.mediawiki#News_files))
* [x] The title of the PR starts with the associated issue number.
* NA I have updated the automated tests and checked that all checks for the PR are green.
* [x] The merge commit will use the below format
  The first line is automatically generated by GitHub based on PR ID and branch name.
  The other lines generated by GitHub should be replaced.

```
Merge pull request #123 from twisted/4356-branch-name-with-issue-id

Author: adiroiban
Reviewer: @twisted/twisted-contributors 
Fixes #11564

Update the development documentation to cover issues and PR handling via GitHub. 
```
